src/shared/conf-parser.c

```diff
@@ -1143,7 +1143,13 @@ int config_parse_mtu(
                            "Failed to parse MTU value '%s', ignoring: %m", rvalue);
                 return 0;
         }
+        if (k == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid MTU value '%s', ignoring: %m", rvalue);
+                return 0;
+        }
 
+        *mtu = k;
```

Comment:
This is not necessary, as `parse_mtu()` already checked its validity. Please drop the second commit.